packages/core/src/tracing/cls.js

```diff
@@ -549,32 +556,79 @@ function skipExitTracing(options) {
       isActive: true,
       extendedResponse: false,
       skipParentSpanCheck: false,
-      skipIsTracing: false
+      skipIsTracing: false,
+      checkReducedSpan: false,
+      skipAllowRootExitSpanPresence: false
     },
     options
   );
 
-  const parentSpan = getCurrentSpan();
+  let isReducedSpan = false;
+  let parentSpan = getCurrentSpan();
+
+  // If there is no active entry span, we fall back to the reduced span of the most recent entry span.
+  // See comment in packages/core/src/tracing/clsHooked/unset.js#storeReducedSpan.
+  if (opts.checkReducedSpan && !parentSpan) {
+    parentSpan = getReducedSpan();
+
+    // We need to remember if a reduced span was used, because for reduced spans
+    // we do NOT trace anymore. The async context got already closed.
+    if (parentSpan) {
+      isReducedSpan = true;
+    }
+  }
+
   const suppressed = tracingSuppressed();
   const isExitSpanResult = isExitSpan(parentSpan);
 
   // CASE: first ask for suppressed, because if we skip the entry span, we won't have a parentSpan
   //       on the exit span, which would create noisy log messages.
   if (suppressed) {
-    if (opts.extendedResponse) return { skip: true, suppressed, isExitSpan: isExitSpanResult };
+    if (opts.extendedResponse) {
+      return { skip: true, suppressed, isExitSpan: isExitSpanResult, parentSpan, allowRootExitSpan };
+    }
+
     return true;
   }
 
+  // DESC: If `allowRootExitSpan` is true, then we have to ignore if there is a parent or not.
+  // NOTE: The feature completely ignores the state of `isTracing`, because
+  //       every exit span would be a separate trace. `isTracing` is always false,
+  //       because we don't have a parent span. The http server span also does not check of `isTracing`,
+  //       because it's the root span.
+  // CASE: Instrumentations can disable the `allowRootExitSpan` feature e.g. loggers.
+  if (!opts.skipAllowRootExitSpanPresence && allowRootExitSpan) {
```

Comment:
Using the same variable name `allowRootExitSpan` will make the instrumentation code more difficult to read as we want the 2 contexts as different.

1 - ROOT EXIT SPAN ENABLED
2-  ROOT EXIT SPAN DISABLED 

We can pass an additional variable to just bypass the checking like above. The check seems like ok to me because we don't want to interfere with the workflow of current instrumentation. So clubbing this check `if (!opts.skipAllowRootExitSpanPresence && allowRootExitSpan) { }` seems relevant to me.

maybe we can change the variable name to a more simple one.

cc: @kirrg001

Reply:
The config `allowRootExitSpan` comes from the tracer configuration level.
Theoretically we could do what you have suggested

> If we need to disable allowRootExitSpan for a particular instrumentation, couldn’t we just pass allowRootExitSpan: false

But I doubt that the readability is not good as Abhilash said

> same variable name allowRootExitSpan will make the instrumentation code more difficult to read

**We want to skip the "root exit span behavior".** We do not want to set the variable to false.